webauthn/helpers/parse_authentication_credential_json.py

```diff
@@ -1,29 +1,96 @@
 import json
-from typing import Callable, Union
-from pydantic import ValidationError
+from json.decoder import JSONDecodeError
+from typing import Union
 
-from .exceptions import InvalidAuthenticationResponse
-from .structs import PYDANTIC_V2, AuthenticationCredential
+from .exceptions import InvalidAuthenticationResponse, InvalidJSONStructure
+from .base64url_to_bytes import base64url_to_bytes
+from .structs import (
+    AuthenticationCredential,
+    AuthenticatorAssertionResponse,
+    AuthenticatorAttachment,
+    PublicKeyCredentialType,
+)
 
 
 def parse_authentication_credential_json(json_val: Union[str, dict]) -> AuthenticationCredential:
     """
     Parse a JSON form of an authentication credential, as either a stringified JSON object or a
     plain dict, into an instance of AuthenticationCredential
     """
-    if PYDANTIC_V2:
-        parsing_method: Callable = AuthenticationCredential.model_validate_json  # type: ignore[attr-defined]
-    else:  # assuming V1
-        parsing_method = AuthenticationCredential.parse_raw
+    if isinstance(json_val, str):
+        try:
+            json_val = json.loads(json_val)
+        except JSONDecodeError:
+            raise InvalidJSONStructure("Unable to decode credential as JSON")
 
-    if isinstance(json_val, dict):
-        json_val = json.dumps(json_val)
+    if not isinstance(json_val, dict):
```

Comment:
In all these .get's - what if the key is missing? possibly this entire parsing block should be in a try... block? or use .get('user_handle', None)...

Reply:
That's why I decided to use `.get()` vs trying to access these properties as strings:

```python
>>> {}['id']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'id'
>>> print({}.get('id'))
None
>>> print({}.get('id', None))
None
```

By default `{}.get()` returns `None` which is why I don't specify `None` for that second argument. Does that answer your question?

Reply:
Duh... back for more coffee...